code/mob/living/critter/small_animal.dm

```diff
@@ -338,6 +338,383 @@ proc/filter_carrier_pets(var/type)
+			src.visible_message(SPAN_COMBAT("<B>[src]</B> tries to bite [target] but \the [target] dodges [pick("nimbly", "effortlessly", "gracefully")]!"))
+			return FALSE
+		src.set_hand(1) //mouth
+		src.set_a_intent(INTENT_HARM)
+		src.hand_attack(target)
+		if (ishuman(target))
+			var/mob/living/carbon/human/human = target
+			var/obj/item/heldthing = human.r_hand ? human.r_hand : human.l_hand ? human.l_hand : null
+			if (prob(10) && heldthing)
+				human.drop_item(heldthing)
+				boutput(target, SPAN_ALERT("[src] bites your hand so hard you drop [heldthing]! [pick("Bad turtle", "Piece of shit", "Ow")]!"))
+		return TRUE
+
+//NOOOOOOO
+	proc/rig_to_explode(mob/user)
+		for (var/mob/living/M in mobs)
+			if (M.mind && M.mind.assigned_role == "Head of Security")
+				boutput(M, SPAN_ALERT("You feel a foreboding feeling about the imminent fate of a certain turtle in [get_area(src)], better act quick."))
+
+		message_admins("[key_name(user)] rigged [src] to explode in [user.loc.loc], [log_loc(user)].")
+		logTheThing(LOG_COMBAT, user, "rigged [src] to explode in [user.loc.loc] ([log_loc(user)])")
+		src.rigged = TRUE
+		src.rigger = user
+
+		var/area/A = get_area(src)
+		if(A?.lightswitch && A?.power_light)
+			src.explode()
+
+	proc/explode()
+		SPAWN(0)
+			src.rigged = FALSE
+			src.rigger = null
+			src.enter_shell()	//enter shell first to give a warning
+			src.exploding = TRUE
+			sleep(0.2 SECONDS)
+			explosion(src, get_turf(src), 0, 1, 2, 2)
+			sleep(4 SECONDS)
+			src.exploding = FALSE
+			var/message = "Check please!"
+			var/chat_text = make_chat_maptext(src, message)
+			for (var/mob/O in all_hearers(7, get_turf(src)))
+				O.show_message("<span class='say bold'>[SPAN_NAME("[src]")]</span> says, [SPAN_MESSAGE("\"[message]\"")]", 2, assoc_maptext = chat_text)
+			playsound(src.loc, 'sound/misc/rimshot.ogg', 50, 1)
+
+	proc/enter_shell()
+		if (src.shell_count) return 0
+		src.shell_count = 10
+		src.ai.disable()
+
+		APPLY_ATOM_PROPERTY(src, PROP_MOB_EXPLOPROT, "turtle_shell", 80)
+		APPLY_ATOM_PROPERTY(src, PROP_MOB_MELEEPROT_HEAD, "turtle_shell", 80)
+		APPLY_ATOM_PROPERTY(src, PROP_MOB_MELEEPROT_BODY, "turtle_shell", 80)
+
+		icon_state = "[base_icon_state]-shell"
+		if (costume_name)
+			src.UpdateOverlays(costume_shell, "costume")
+		density = TRUE
+		UpdateIcon()
+		src.visible_message(SPAN_ALERT("<b>[src]</b> retreats into [his_or_her(src)] shell!"))
+		return 1
+
+	proc/exit_shell()
+
+		src.shell_count = 0
+		src.ai.enable()
+
+		REMOVE_ATOM_PROPERTY(src, PROP_MOB_EXPLOPROT, "turtle_shell")
+		REMOVE_ATOM_PROPERTY(src, PROP_MOB_MELEEPROT_HEAD, "turtle_shell")
+		REMOVE_ATOM_PROPERTY(src, PROP_MOB_MELEEPROT_BODY, "turtle_shell")
+
+		icon_state = base_icon_state
+		if (costume_name)
+			src.UpdateOverlays(costume_alive, "costume")
+		density = FALSE
+		UpdateIcon()
+		src.visible_message(SPAN_NOTICE("<b>[src]</b> comes out of [his_or_her(src)] shell!"))
+		return 1
+
+
+	proc/give_beret(var/obj/hat, var/mob/user)
```

Comment:
Sylvester seems to revive with his beret sprite, which is I think what this is for, should probably check that on revive somehow?

Reply:
Im not sure what gets called where when a mob is revived (e.g. where it sets the icon back to the initial one) but it should definitely call UpdateIcon()